diff --git a/src/common/GameValues.cpp b/src/common/GameValues.cpp
index ec8ae6e1..8c8532f7 100644
--- a/src/common/GameValues.cpp
+++ b/src/common/GameValues.cpp
@@ -62,6 +62,22 @@ SDL_KEYTYPE controlkeys[2][2][4][NUM_KEYS] = { { { {SDLK_LEFT, SDLK_RIGHT, SDLK_
         }
     }
 };
+#elif defined(__SWITCH__)
+    //left, right, jump, down, turbo, powerup, start, cancel;
+    { { {JOY_STICK_1_LEFT, JOY_STICK_1_RIGHT, JOY_BUTTON_START, JOY_STICK_1_DOWN, JOY_BUTTON_START + 1, JOY_BUTTON_START + 2, JOY_BUTTON_START + 10, JOY_BUTTON_START + 11},
+            {JOY_STICK_1_LEFT, JOY_STICK_1_RIGHT, JOY_BUTTON_START, JOY_STICK_1_DOWN, JOY_BUTTON_START + 1, JOY_BUTTON_START + 2, JOY_BUTTON_START + 10, JOY_BUTTON_START + 11},
+            {JOY_STICK_1_LEFT, JOY_STICK_1_RIGHT, JOY_BUTTON_START, JOY_STICK_1_DOWN, JOY_BUTTON_START + 1, JOY_BUTTON_START + 2, JOY_BUTTON_START + 10, JOY_BUTTON_START + 11},
+            {JOY_STICK_1_LEFT, JOY_STICK_1_RIGHT, JOY_BUTTON_START, JOY_STICK_1_DOWN, JOY_BUTTON_START + 1, JOY_BUTTON_START + 2, JOY_BUTTON_START + 10, JOY_BUTTON_START + 11}
+        },
+
+        //up, down, left, right, select, cancel, random, fast scroll
+        {   {JOY_STICK_1_UP, JOY_STICK_1_DOWN, JOY_STICK_1_LEFT, JOY_STICK_1_RIGHT, JOY_BUTTON_START, JOY_BUTTON_START + 1, JOY_BUTTON_START + 2, JOY_BUTTON_START + 3},
+            {JOY_STICK_1_UP, JOY_STICK_1_DOWN, JOY_STICK_1_LEFT, JOY_STICK_1_RIGHT, JOY_BUTTON_START, JOY_BUTTON_START + 1, JOY_BUTTON_START + 2, JOY_BUTTON_START + 3},
+            {JOY_STICK_1_UP, JOY_STICK_1_DOWN, JOY_STICK_1_LEFT, JOY_STICK_1_RIGHT, JOY_BUTTON_START, JOY_BUTTON_START + 1, JOY_BUTTON_START + 2, JOY_BUTTON_START + 3},
+            {JOY_STICK_1_UP, JOY_STICK_1_DOWN, JOY_STICK_1_LEFT, JOY_STICK_1_RIGHT, JOY_BUTTON_START, JOY_BUTTON_START + 1, JOY_BUTTON_START + 2, JOY_BUTTON_START + 3}
+        }
+    }
+};
 #else
     //left, right, jump, down, turbo, powerup, start, cancel;
     { { {JOY_STICK_1_LEFT, JOY_STICK_1_RIGHT, JOY_BUTTON_START, JOY_STICK_1_DOWN, JOY_BUTTON_START + 1, JOY_BUTTON_START + 2, JOY_BUTTON_START + 3, JOY_BUTTON_START + 4},
@@ -229,7 +245,7 @@ void CGameValues::init()
         }
 
         //Set the players input to the default configuration (will be overwritten by options.bin settings)
-#ifdef _XBOX
+#if defined(_XBOX) || defined(__SWITCH__)
         inputConfiguration[iPlayer][1].iDevice = iPlayer;
         playerInput.inputControls[iPlayer] = &inputConfiguration[iPlayer][1];
 #else
@@ -446,7 +462,7 @@ void CGameValues::ReadBinaryConfig() {
         for (short iPlayer = 0; iPlayer < MAX_PLAYERS; iPlayer++) {
             short iDevice = controls.read_i16();
 
-#ifdef _XBOX
+#if defined(_XBOX) || defined(__SWITCH__)
             playerInput.inputControls[iPlayer] = &inputConfiguration[iPlayer][1]; //Always use gamepads as input devices on xbox
 #else
             if (iDevice >= joystickcount)
diff --git a/src/common/input.cpp b/src/common/input.cpp
index b6011f1f..dfcc4840 100644
--- a/src/common/input.cpp
+++ b/src/common/input.cpp
@@ -21,7 +21,7 @@ CPlayerInput::CPlayerInput()
 
 void CPlayerInput::CheckIfMouseUsed()
 {
-#ifndef _XBOX
+#if !defined(_XBOX) && !defined(__SWITCH__)
 
     for (short iGameState = 0; iGameState < 2; iGameState++) {
         for (short iPlayer = 0; iPlayer < MAX_PLAYERS; iPlayer++) {
@@ -56,7 +56,7 @@ void CPlayerInput::ClearPressedKeys(short iGameState)
         for (int iKey = 0; iKey < NUM_KEYS; iKey++) {
 			outputControl->keys[iKey].fPressed = false;
 
-#ifndef _XBOX
+#if !defined(_XBOX) && !defined(__SWITCH__)
             if (fUsingMouse) {
 				SDL_KEYTYPE iInputKey = inputControl->keys[iKey];
 
diff --git a/src/smw/ui/MI_InputControlContainer.cpp b/src/smw/ui/MI_InputControlContainer.cpp
index 460da88b..43c1fa3c 100644
--- a/src/smw/ui/MI_InputControlContainer.cpp
+++ b/src/smw/ui/MI_InputControlContainer.cpp
@@ -93,6 +93,12 @@ MI_InputControlField::~MI_InputControlField()
         "Pad Left", "Pad Right", "A Button", "B Button", "X Button", "Y Button", "Black Button", "White Button", "Left Trigger", "Right Trigger",
         "Start Button", "Back Button", "Left Stick Click", "Right Stick Click", "Button 1", "Button 2", "Button 3", "Button 4", "Button 5", "Button 6"
     };
+#elif defined(__SWITCH__)
+    const char * MI_InputControlField::Joynames[30] = {
+        "Left Stick Up", "Left Stick Down", "Left Stick Left", "Left Stick Right", "Right Stick Up", "Right Stick Down", "Right Stick Left", "Right Stick Right", "Button 1", "Button 2",
+        "Button 3", "Button 4", "A Button", "B Button", "X Button", "Y Button", "Left Stick Button", "Right Stick Button", "L Trigger", "R Trigger",
+        "ZL Trigger", "ZR Trigger", "Plus Button", "Minus Button", "D-Pad Left", "D-Pad Up", "D-Pad Right", "D-Pad Down", "Button 5", "Button 6"
+    };
 #else
     const char * MI_InputControlField::Joynames[30] = {
         "Joystick Up", "Joystick Down", "Joystick Left", "Joystick Right", "Stick 2 Up", "Stick 2 Down", "Stick 2 Left", "Stick 2 Right", "Pad Up", "Pad Down",
